platform/linuxbsd/x11/display_server_x11.cpp

```diff
@@ -5425,6 +5475,268 @@ DisplayServer::VSyncMode DisplayServerX11::window_get_vsync_mode(WindowID p_wind
+		}
+
+		// Second, for each screen, check if the desired rectangle is within a portion of the screen
+		// that is outside the working area. Each screen can have a different working area
+		// depending on top, bottom, or side panels.
+		int desired_area = desired_rect.get_area();
+		int count = get_screen_count();
+		for (int i = 0; i < count; i++) {
+			Rect2i screen_rect = _screen_get_rect(i);
+			if (screen_rect.intersection(desired_rect).get_area() == 0) {
+				continue;
+			}
+
+			// The desired rect is inside this screen.
+			Rect2i screen_usable_rect = screen_get_usable_rect(i);
+			int screen_usable_area = screen_usable_rect.intersection(desired_rect).get_area();
+			if (screen_usable_area == desired_area) {
+				// The desired rect is fulling inside the usable rect of the screen. No need to resize.
+				continue;
+			}
+
+			if (desired_rect.position.x >= screen_rect.position.x && desired_rect.position.x < screen_usable_rect.position.x) {
+				int offset = screen_usable_rect.position.x - desired_rect.position.x;
+				desired_rect.size.x = MAX(desired_rect.size.x - offset, 0);
+				desired_rect.position.x += offset;
+			}
+			if (desired_rect.position.y >= screen_rect.position.y && desired_rect.position.y < screen_usable_rect.position.y) {
+				int offset = screen_usable_rect.position.y - desired_rect.position.y;
+				desired_rect.size.y = MAX(desired_rect.size.y - offset, 0);
+				desired_rect.position.y += offset;
+			}
+
+			Vector2i desired_end = desired_rect.get_end();
+			Vector2i screen_end = screen_rect.get_end();
+			Vector2i screen_usable_end = screen_usable_rect.get_end();
+			if (desired_end.x > screen_usable_end.x && desired_end.x <= screen_end.x) {
+				desired_rect.size.x = MAX(desired_rect.size.x - (desired_end.x - screen_usable_end.x), 0);
+			}
+			if (desired_end.y > screen_usable_end.y && desired_end.y <= screen_end.y) {
+				desired_rect.size.y = MAX(desired_rect.size.y - (desired_end.y - screen_usable_end.y), 0);
+			}
+		}
+
+		if (desired_rect.size.x < 100 || desired_rect.size.y < 100) {
+			p_visible = false;
+		}
+
+		if (p_visible) {
+			Rect2i current_process_window_rect = _get_window_rect(ep->process_window);
+			if (current_process_window_rect != desired_rect) {
+				DEBUG_LOG_X11("Embedding XMoveResizeWindow process %ld, window %lu to %d, %d, %d, %d \n", p_pid, wd.x11_window, desired_rect.position.x, desired_rect.position.y, desired_rect.size.x, desired_rect.size.y);
+				XMoveResizeWindow(x11_display, ep->process_window, desired_rect.position.x, desired_rect.position.y, desired_rect.size.x, desired_rect.size.y);
+			}
+		}
+	}
+
+	if (ep->visible != p_visible) {
+		if (p_visible) {
+			XMapWindow(x11_display, ep->process_window);
+		} else {
+			XUnmapWindow(x11_display, ep->process_window);
+		}
+		ep->visible = p_visible;
+	}
+
+	if (p_grab_focus) {
+		XSetInputFocus(x11_display, ep->process_window, RevertToParent, CurrentTime);
```

Comment:
This call constantly fails and spam BadMatch: Opcode 42 errors. Probably should be delayed or called from window mapped callback.

Reply:
I was able to reproduce the issue once in a while. The issue seems related to `XSetInputFocus` to soon after the window was mapped sometimes. I added a check if the window is `viewable` before calling `XSetInputFocus` and if it's not the current window that has the input focus.

Also, I found an issue where p_grab_focus was passed at true when the embedded window was hidden or when the editor does not have the focus. That caused the issue where XSetInputFocus` was called on an hidden window  and caused the embedded window to grab the focus even if the user focused another app while the game was starting.